server/index.js

```diff
@@ -33,6 +33,23 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const sql = `
+    select *
+      from "products"
+     where "productId" = $1
+  `;
+  const params = [req.params.productId];
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        next(new ClientError(`Unable to find  id of ${params[0]}`), 404);
+      }
+      res.json(result.rows[0]);
```

Comment:
Be sure to include an `else` or `return` here so that your server doesn't try to respond twice.